src/solver/infeasible-problem-analysis/problem.cpp

```diff
@@ -11,80 +11,179 @@ namespace Antares
             }
         }
     }
 }
 
-void InfeasibleProblemAnalysis::buildObjective() const
+void SlackVariablesAnalysis::buildObjective() const
 {
-    MPObjective* objective = mSolver->MutableObjective();
+    MPObjective* objective = problem_->MutableObjective();
     // Reset objective function
     objective->Clear();
     // Only slack variables have a non-zero cost
-    for (const MPVariable* slack : mSlackVariables)
+    for (const MPVariable* slack : slackVariables_)
     {
         objective->SetCoefficient(slack, 1.);
     }
     objective->SetMinimization();
 }
 
-MPSolver::ResultStatus InfeasibleProblemAnalysis::Solve() const
+MPSolver::ResultStatus SlackVariablesAnalysis::Solve() const
 {
-    return mSolver->Solve();
+    return problem_->Solve();
 }
 
-InfeasibleProblemReport InfeasibleProblemAnalysis::produceReport()
+void SlackVariablesAnalysis::printReport()
 {
-    addSlackVariables();
-    if (mSlackVariables.empty())
+    InfeasibleProblemReport report(slackVariables_);
+    report.prettyPrint();
+}
+
+// ============================
+// Variables bounds analysis
+// ============================
+
+void VariablesBoundsAnalysis::run()
+{
+    for (auto& var : problem_->variables())
     {
-        throw SlackVariablesEmpty(
-          "Cannot generate infeasibility report: no constraints have been selected");
+        double lowBound = var->lb();
+        double upBound = var->ub();
+        std::string name = var->name();
+        if (lowBound > upBound)
+        {
+            storeIncorrectVariable(name, lowBound, upBound);
+        }
     }
-    buildObjective();
-    const MPSolver::ResultStatus status = Solve();
-    if ((status != MPSolver::OPTIMAL) && (status != MPSolver::FEASIBLE))
+
+    if (foundIncorrectVariables())
+        hasDetectedInfeasibilityCause_ = true;
+}
+
+void VariablesBoundsAnalysis::storeIncorrectVariable(std::string name, double lowBound, double upBound)
+{
+    incorrectVars_.push_back(VariableBounds(name, lowBound, upBound));
+}
+
+bool VariablesBoundsAnalysis::foundIncorrectVariables()
+{
+    return (incorrectVars_.size() != 0);
+}
+
+void VariablesBoundsAnalysis::printReport()
+{
+    for (auto& var : incorrectVars_)
+    {
+        logs.notice() << var.name << " : low bound = " << var.lowBound << ", up bound = " << var.upBound;
+    }
+}
+
+
+// ===============================
+// Unfeasibility analyzer 
+// ===============================
+
+// gp : this class should be renamed into UnfeasibilityAnalyzer
+
+InfeasibleProblemAnalysis::InfeasibleProblemAnalysis(const std::string& solverName, const PROBLEME_SIMPLEXE_NOMME* ProbSpx)
+{
+    // gp : Here we have a dependency on PROBLEME_SIMPLEXE_NOMME and MPSolver.
+    // gp : We should Convert() the PROBLEME_SIMPLEXE_NOMME into a MPSolver outside (and before) this constructor.
+    // gp : And so we should have a MPSolver* passed here.
+    // gp : It would be easier to test this class in isolation.
```

Comment:
I agree, but it's actually quite convenient to have this intermediate layer (probleme_simplexe), since class MPSolver has deleted copy constructor & operator.

If we passed the original problem, we'd need to make a copy. Although it is possible, it can be quite time-consuming.

Reply:
For more flexibility in the use of this class and to break the dependency, we can use a supplier of MPSolver as an argument: `std::function<std::shared_ptr<MPSolver>()>`.

We can then use it either with a raw MPSolver, in unit tests, or with a PROBLEME_SIMPLEXE_NOMME that will be converted as needed (by a lambda for ex):

```cpp
PROBLEME_SIMPLEXE_NOMME* problem = ...;
InfeasibleProblemAnalysis analysis("XPRESS", [problem]() {
    return std::shared_ptr<MPSolver>(ProblemSimplexeNommeConverter(solverName, ProbSpx).Convert()
});
```

Reply:
@flomnes : I still think that creating a **MPSolver** from a **PROBLEME_SIMPLEXE_NOMME** is not the concern of the unfeasible problem analyzer. This analyzer should only receive a optimization problem and handle several analysis on this problem. So I would recommend to pass to the constructor at most a problem (see comment below for a change on this).

@sylvlecl : We talked about your suggestion. It's useful if we have to regenerate a fresh copy of the original problem in case an analysis modified this problem (as it's the case for slack variables analysis), and another analysis has to do its work after that.
But it's not the case for now. I would reserve the fresh copy creation handling for a possible future change that really requires it. For now we respect **YAGNI**.